### PR TITLE
Clean also ignored files during VMR syncs

### PR DESCRIPTION
**Pull request by premun**
Found the problem here https://github.com/dotnet/installer/issues/19307

### Release Note Category
- [ ] Feature changes/additions 
- [x] Bug fixes
- [ ] Internal Infrastructure Improvements

### Release Note Description
Clean ignored files properly during VMR syncs

### CODE REVIEW

**Review by oleksandr-didyk — APPROVED** [review]

File: src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs [hunk 1]
```diff
@@ -110,7 +110,7 @@ public async Task ResetWorkingTree(NativePath repoPath, UnixPath? relativePath =
         }
 
         // Also remove untracked files (in case files were removed in index)
-        result = await _processManager.ExecuteGit(repoPath, ["clean", "-df", relativePath], cancellationToken: CancellationToken.None);
+        result = await _processManager.ExecuteGit(repoPath, ["clean", "-xdf", relativePath], cancellationToken: CancellationToken.None);
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs:113]: FYI `-xdff` would remove nested submodules that got deleted too, it's been my go to git clean command :)

Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs:113]: Ah, thanks. Fortunately, there's no submodules :)

Comment (akoeplinger) [reply src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs:113]: right now 😆

Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs:113]: But I see what you mean

**Pull request merged**